src/client/Microsoft.Identity.Client/TelemetryCore/Internal/Events/ApiEvent.cs

```diff
@@ -63,164 +35,102 @@ public enum ApiIds
         public ApiEvent(
             ICoreLogger logger,
             ICryptographyManager cryptographyManager,
-            string correlationId) : base(EventNamePrefix + "api_event", correlationId)
+            Guid correlationId)
         {
             _logger = logger;
-            _cryptographyManager = cryptographyManager;            
-        }
-
-        public ApiTelemetryId ApiTelemId
-        {
-            set => this[MsalTelemetryBlobEventNames.ApiTelemIdConstStrKey] = ((int) value).ToString(CultureInfo.InvariantCulture);
+            _cryptographyManager = cryptographyManager;
+            CorrelationId = correlationId;
         }
 
-        public ApiIds ApiId
-        {
-            get => TryGetValue(MsalTelemetryBlobEventNames.ApiIdConstStrKey, out string apiIdString) ? (ApiIds)Enum.Parse(typeof(ApiIds), apiIdString) : ApiIds.None;
+        public Guid CorrelationId { get; set; }
 
-            set => this[MsalTelemetryBlobEventNames.ApiIdConstStrKey] = ((int) value).ToString(CultureInfo.InvariantCulture);
-        }
+        public ApiIds ApiId { get; set; }
 
         public string ApiIdString
         {
-            get => ContainsKey(MsalTelemetryBlobEventNames.ApiIdConstStrKey) ? 
-                this[MsalTelemetryBlobEventNames.ApiIdConstStrKey] : 
-                null;
+            get => ((int)ApiId).ToString(CultureInfo.InvariantCulture);
         }
 
-        public string TokenEndpoint
-        {
-            get; set;
-        }
+        public string TokenEndpoint { get; set; }
 
-        public Uri Authority
-        {
-            set => this[AuthorityKey] = ScrubTenant(value)?.ToLowerInvariant();
-        }
+        // Some of these properties like Authority, TenantId, LoginHint, etc.
```

Comment:
This is an internal class. If they are not being used, it will be safe to remove.

Reply:
Fixed in PR #3058.